src/index.js

```diff
@@ -159,9 +160,9 @@ const GravityFormForm = ({
                                 id={`gform_submit_button_${id}`}
                                 type="submit"
                             >
-                                {formLoading ? (
+                                {formLoading && buttonLoading ? (
                                     <span className="gravityform__button__loading_span">
```

Comment:
In your PR you mention the loading could be text or a component. Can you do a check, so that if it is a component it isnt wrapped in this span?